src/popup/components/Header.vue

```diff
@@ -85,60 +91,46 @@ export default {
     BtnPlain,
     BtnIcon,
   },
-  subscriptions() {
-    return {
-      superheroNotifications: this.$store.state.observables.notifications,
-    };
-  },
-  computed: {
-    ...mapGetters(['isLoggedIn']),
-    ...mapState(['notifications', 'pageTitle']),
-    truncateStr() {
-      const { pageTitle, title } = this;
-      return pageTitle || (title ? this.$t(`pages.titles.${title}`) : '');
-    },
-    title() {
-      return this.$route.meta.title;
-    },
-    isModalHeader() {
-      return this.$route.meta.isModalHeader;
-    },
-    showBack() {
-      return (
-        (this.$route.meta.backButton !== undefined ? this.$route.meta.backButton : true)
-        && this.title
-      );
-    },
-    showMore() {
-      return this.$route.name !== 'more' && !this.$route.meta?.closeButton;
-    },
-    showNotifications() {
-      return this.$route.name !== 'notifications' && !this.$route.meta?.hideNotificationsIcon;
-    },
-    isLogoDisabled() {
-      return this.$route.name === 'account';
-    },
-  },
-  methods: {
-    back() {
-      if (this.$store.state.route.from.path === '/') {
-        this.$router.push(this.isLoggedIn ? '/account' : '/');
-        return;
+  setup(props, { root }) {
+    const isLoggedIn = useGetter('isLoggedIn');
+    const routeMeta = computed(() => root.$route.meta as WalletRouteMeta);
+    const showHeaderNavigation = computed(() => !!routeMeta.value?.showHeaderNavigation);
+    const isLogoDisabled = computed(() => root.$route.name === ROUTE_ACCOUNT);
+    const titleTruncated = computed(
+      () => routeMeta.value?.title ? root.$t(`pages.titles.${routeMeta.value.title}`) : '',
+    );
+
+    function back() {
+      if (root.$store.state.route.from.name === ROUTE_INDEX) {
+        return root.$router.push({
+          name: (isLoggedIn.value) ? ROUTE_ACCOUNT : ROUTE_INDEX,
+        });
       }
-      if (this.$route.meta?.backRoute) {
+      if (root.$route.meta?.backRoute) {
         // TODO: rewrite back button logic in more unified way
-        this.$router.push(this.$route.meta?.backRoute);
-        return;
+        return root.$router.push(root.$route.meta?.backRoute);
       }
-      this.$router.back();
-    },
-    close() {
-      this.$router.replace(
-        this.isLoggedIn ? '/account' : '/',
-      );
-    },
+      return root.$router.back();
+    }
+
+    function close() {
+      root.$router.replace({
+        name: (isLoggedIn.value) ? ROUTE_ACCOUNT : ROUTE_INDEX,
+      });
```

Comment:
this code is used above - maybe you could extract it to a computed?

Reply:
OK, changed.